module/friendship/adapter/postgres/repository/subscription_test.go

```diff
@@ -0,0 +1,390 @@
+
+			subIds := make([]string, 0)
+			sub := domain.Subscription{
+				UserID:       util.GenUUID(),
+				SubscriberID: util.GenUUID(),
+				Status:       domain.SubscriptionStatusSubscribed,
+			}
+			suite.prepareSubscription(t, ctx, sub)
+			var err error
+			id := "fake-subscription"
+			if tc.isFounded {
+				id, err = repo.Create(ctx, sub)
+				assert.NoError(t, err)
+				sub.Id = id
+				subIds = append(subIds, id)
+			}
+			mentionedEmail := make([]string, 0)
+			if len(tc.mentionedEmails) > 0 {
+				for _, email := range tc.mentionedEmails {
+					newEmail := mapEmailUser[email].Email
+					if newEmail != "" {
+						mentionedEmail = append(mentionedEmail, newEmail)
+					}
+				}
+			}
+
+			if len(tc.blockedEmails) > 0 {
+				for _, email := range tc.blockedEmails {
+					id, err = repo.UnsertSubscription(ctx, domain.Subscription{
+						UserID:       sub.UserID,
+						SubscriberID: mapEmailUser[email].ID,
+						Status:       domain.SubscriptionStatusUnsubscribed,
+					})
+					assert.NoError(t, err)
+					subIds = append(subIds, id)
+				}
+			}
+
+			result, err := repo.GetSubscriptionEmailsByUserIDAndEmails(ctx, sub.UserID, mentionedEmail)
+			assert.NoError(t, err)
+			if tc.isFounded {
+				assert.Equal(t, len(tc.result)+1, len(result))
+				assert.True(t, util.IsContain(result, sub.SubscriberID+"@example.com"))
+
+				for _, v := range tc.result {
+					assert.True(t, util.IsContain(result, mapEmailUser[v].ID+v))
+				}
+			} else {
+				assert.Len(t, result, 0)
+			}
+
+			suite.rollbackSubscription(t, ctx, sub, subIds)
+		})
+
+	}
+}
+
+func (g *Suite) prepareSubscription(t *testing.T, ctx context.Context, sub domain.Subscription) {
+	db := g.db.Model(ctx)
+	u := model.User{
+		ID:    sub.UserID,
+		Email: sub.UserID + "@example.com",
+	}
+	err := u.Insert(ctx, db, boil.Infer())
+	assert.NoError(t, err)
+
+	u = model.User{
+		ID:    sub.SubscriberID,
+		Email: sub.SubscriberID + "@example.com",
+	}
+	err = u.Insert(ctx, db, boil.Infer())
+	assert.NoError(t, err)
+}
+
+func (g *Suite) initialUsers(t *testing.T, ctx context.Context, emails []string) map[string]model.User {
+	emails = util.RemoveDuplicates(emails)
+	result := make(map[string]model.User, 0)
+	for _, email := range emails {
+		randString := util.GenUUID()
+		user := model.User{ID: randString, Email: randString + email}
+		err := user.Insert(ctx, g.db.Model(ctx), boil.Infer())
+		assert.NoError(t, err)
+		result[email] = user
+	}
+	return result
+}
+
+func (g *Suite) rollbackSubscription(t *testing.T, ctx context.Context, sub domain.Subscription, subIds []string) {
+	db := g.db.Model(ctx)
+	ids := util.RemoveDuplicates(subIds)
+	if len(ids) > 0 {
+		for _, id := range ids {
+			sub2 := &model.Subscription{
+				ID: id,
+			}
+			_, err := sub2.Delete(ctx, db)
+			assert.NoError(t, err)
+		}
+	}
+
+	users := &model.UserSlice{
+		{ID: sub.UserID},
+		{ID: sub.SubscriberID},
+	}
+	_, err := users.DeleteAll(ctx, db)
+	assert.NoError(t, err)
+}
```

Comment:
if this function is reusable in many cases, can we make it a common function ?

Reply:
Nope anh, because this function only use for Subscription Business, but a common function is only for Technical that is independent with Business, so we shouldn't make the above function as a common function.